README.md

```diff
@@ -1,93 +1,93 @@
+It parses the expiration time of your access token and checks to see if it is expired before every request. If it has expired, a request to
 refresh and store a new access token is automatically performed before the request proceeds.
 
 ## How do I use it?
 
-- Define a token refresh function
-- Configure the interceptor
-- Store tokens on login with `setAuthTokens()`
+1. Create an axios instance
+2. Define a token refresh function
+3. Configure the interceptor
+4. Store tokens on login with `setAuthTokens()`
+5. Clear tokens on logout with `clearAuthTokens()`
 
-### Apply interceptor:
+### Applying the interceptor
 
 ```typescript
-import { IAuthTokens, TokenRefreshRequest, useAuthTokenInterceptor } from 'axios-jwt'
+// api.ts
+
+import { IAuthTokens, TokenRefreshRequest, applyAuthTokenInterceptor } from 'axios-jwt'
 import axios from 'axios'
-// your axios instance that you wish to apply the interceptor to
-import apiClient from '../apiClient'
 
-const BASE_URL = process.env.REACT_APP_BASE_URL
-if (!BASE_URL) throw new Error('BASE_URL is not defined')
+const BASE_URL = 'https://api.example.com'
 
-// type of response from login endpoint
-export interface IAuthResponse {
-  access_token: string
-  refresh_token: string
-}
+// 1. Create an axios instance that you wish to apply the interceptor to
+export const apiClient = axios.create({ baseURL: BASE_URL })
 
-// refresh token endpoint
-const refreshEndpoint = `${BASE_URL}/auth/refresh_token`
+// 2. Define token refresh function.
+const requestRefresh: TokenRefreshRequest = async (refreshToken: string): Promise<string> => {
 
-// transform response into IAuthTokens
-// this assumes your auth endpoint returns `{"access_token": ..., "refresh_token": ...}`
-export const authResponseToAuthTokens = (res: IAuthResponse): IAuthTokens => ({
-  accessToken: res.access_token,
-  refreshToken: res.refresh_token,
-})
+  // Important! Do NOT use the axios instance that you supplied to applyAuthTokenInterceptor (in our case 'apiClient')
+  // because this will result in an infinite loop when trying to refresh the token.
+  // Use the global axios client or a different instance
+  const response = await axios.post(`${BASE_URL}/auth/refresh_token`, { token: refreshToken })
 
-// define token refresh function
-const requestRefresh: TokenRefreshRequest = async (refreshToken: string): Promise<string> => {
-  // perform refresh
-  return (await axios.post(refreshEndpoint, { token: refreshToken })).data.access_token
+  return response.data.access_token
 }
 
-// add interceptor to your axios instance
-useAuthTokenInterceptor(apiClient, { requestRefresh })
+// 3. Add interceptor to your axios instance
+applyAuthTokenInterceptor(apiClient, { requestRefresh })
 ```
 
-### Login/logout:
+### Login/logout
 
 ```typescript
+// login.ts
+
 import { isLoggedIn, setAuthTokens, clearAuthTokens, getAccessToken, getRefreshToken } from 'axios-jwt'
+import { apiClient } from '../apiClient'
```

Comment:
small nit. I prefer seeing imports with `@`, but this is `README.md` so your call.

Reply:
not sure what you mean here